### PR TITLE
[PLGN-628] Service Now- Update Triggers

### PR DESCRIPTION
**Pull request by jerichardson-r7**
### Description

Describe the proposed changes:

  - Update triggers to not force the user to enter a system_id or allow them to enter a list of system_ids
  - Updated connection version
  - Fixed Snyk Vulnerability
  https://app.snyk.io/org/soar-insightconnect-new/project/063bfd6b-2987-4fa3-b4e2-af1da89dd802
  
  https://rapid7.atlassian.net/browse/SOAR-16881

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code
- [ ] Manual testing

### CODE REVIEW

File: plugins/servicenow/icon_servicenow/triggers/vulnerability_updated/trigger.py [hunk 1]
```diff
@@ -18,48 +17,165 @@ def __init__(self):
...
+
+        :return: A generator that yields a tuple consisting of a string and a dictionary.
+        :rtype: Generator[Tuple[str, Dict[str, Any]], Any, None]
+        """
+
+        return ((vul.pop("sys_id", ""), vul) for vul in list_of_vuls)
+
+
+
+    # def run(self, params={}):
```
Comment (ablakley-r7) [inline plugins/servicenow/icon_servicenow/triggers/vulnerability_updated/trigger.py:134]: Can we remove this old code?

File: plugins/servicenow/plugin.spec.yaml [hunk 2]
```diff
@@ -4,8 +4,8 @@ products: ["insightconnect"]
 name: servicenow
 title: ServiceNow
 description: ServiceNow is a tool for managing incidents and configuration management. Using the ServiceNow plugin for Rapid7 InsightConnect, users can manage all aspects of incidents including creation, search, updates, as well as monitor them for changes
-version: 7.4.1
-connection_version: 7
```
Comment (ablakley-r7) [inline plugins/servicenow/plugin.spec.yaml:8]: Should this stay at 7?

File: plugins/servicenow/help.md [hunk 3]
```diff
@@ -1681,6 +1685,7 @@ Example output:
 
 # Version History
 
+* 8.0.0 - `Incident Created`: Resolved issue related to trigger not working. Updated SDK
```
Comment (ablakley-r7) [inline plugins/servicenow/help.md:1688]: I think the text is duplicated from 7.4.1

**Review by ablakley-r7 — CHANGES_REQUESTED** [review]
Looks good. Can the unit tests be reviewed? They are passing but I think the existing tests for the updated triggers should fail since changes to input/output was made.

**Review by igorski-r7 — COMMENTED** [review]
Same comments as @ablakley-r7 mentioned 😄

File: plugins/servicenow/help.md (hunk 3 above)
Comment (jerichardson-r7) [reply plugins/servicenow/help.md:1688]: Ah man, thanks for spotting this as i've just noticed it hasn't pulled in some of my changes!

File: plugins/servicenow/plugin.spec.yaml (hunk 2 above)
Comment (jerichardson-r7) [reply plugins/servicenow/plugin.spec.yaml:8]: We have decided to move up the connection version after the last trigger change 🙂

**Review by ablakley-r7 — APPROVED** [review]

**Review by igorski-r7 — APPROVED** [review]

**Pull request merged**